--- a/Cargo.toml
+++ b/Cargo.toml
@@ -2,14 +2,14 @@
 name = "brush-parser"
 description = "POSIX/bash shell tokenizer and parsers (used by brush-shell)"
 version = "0.4.0"
-authors.workspace = true
-categories.workspace = true
-edition.workspace = true
-keywords.workspace = true
-license.workspace = true
-readme.workspace = true
-repository.workspace = true
-rust-version.workspace = true
+authors = ["reuben olinsky"]
+categories = ["command-line-utilities", "development-tools"]
+edition = "2024"
+keywords = ["cli", "shell", "sh", "bash", "script"]
+license = "MIT"
+readme = false
+repository = "https://github.com/reubeno/brush"
+rust-version = "1.88.0"
 
 [lints]
 workspace = true
